test/config-luatest/gh_10304_allow_specific_function_access_test.lua

```diff
@@ -0,0 +1,70 @@
+local fun = require('fun')
+local yaml = require('yaml')
+local t = require('luatest')
+local treegen = require('test.treegen')
+local server = require('test.luatest_helpers.server')
+local helpers = require('test.config-luatest.helpers')
+
+local g = helpers.group()
+
+g.test_lua_call_func = function(g)
+    local dir = treegen.prepare_directory(g, {}, {})
+    local config = {
+        credentials = {
+            users = {
+                guest = {
+                    roles = {'super'},
+                },
+                alice = {
+                    password = "ALICE",
+                    privileges = {
+                        {
+                            permissions = {"execute"},
+                            lua_call = {"foo", "bar"}
+                        },
+                    }
+                },
+            },
+        },
+        iproto = {
+            listen = {{uri = 'unix/:./{{ instance_name }}.iproto'}},
+        },
+        groups = {
+            ['group-001'] = {
+                replicasets = {
+                    ['replicaset-001'] = {
+                        instances = {
+                            ['instance-001'] = {},
+                        },
+                    },
+                },
+            },
+        },
+    }
+
+    local config_file = treegen.write_script(dir, 'config.yaml',
+                                             yaml.encode(config))
+    local opts = {config_file = config_file, chdir = dir}
+    g.server = server:new(fun.chain(opts, {alias = 'instance-001'}):tomap())
+    g.server:start()
+    g.server:exec(function()
+        local netbox = require('net.box')
+        box.cfg{listen=3303}
+        local con = netbox.connect(3303, {user='alice', password='ALICE'})
```

Comment:
Please, don't use a hardcoded TCP port number in a test. If the port is used by another process (for example, another test), this test will fail. It makes the testing unstable.

This tarantool instance already listens on a unix socket. Let's use it. You can use `config:instance_uri()` ([docs](https://www.tarantool.io/en/doc/latest/reference/reference_lua/config/#config-api-reference-instance-uri)) for convenience.

Or even `connpool.call()` ([docs](https://www.tarantool.io/en/doc/latest/reference/reference_lua/connpool/#connpool-module-call)) if you set `iproto.advertise.peer.login` to `'alice'`.

Just net.box is OK as well.

Reply:
port replaced with `config:instance_uri()`